src/trade-params/FullTradeParams.js

```diff
@@ -94,13 +94,19 @@ export default class FullTradeParams extends Component {
     componentWillMount() {
         this.onAssetChange();
     }
-
+    componentDidMount() {
+        const { actions, index, trade } = this.props;
+        if (!trade.hasOwnProperty('proposal')) {
```

Comment:
Really no need to overdo the hasOwnProperty.
There is no prototype chain here.
if (!trade.proposal) is enough